File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int idx1 = 0, idx2 = 0, n1 = size(s), n2 = size(t);
        while (idx1 < n1 and idx2 < n2) {
            if (s[idx1] == t[idx2])
                ++idx1, ++idx2;
            else 
                ++idx2;
        }
        return idx1 == n1;
    }
};

File: 0392-is-subsequence/0392-is-subsequence.ts
function isSubsequence(s: string, t: string): boolean {
    let idx1: number = 0, idx2: number = 0;
    const n1: number = s.length, n2: number = t.length;
        while (idx1 < n1 && idx2 < n2) {
            if (s[idx1] == t[idx2])
                ++idx1, ++idx2;
            else 
                ++idx2;
        }
        return idx1 == n1;
};